### PR TITLE
Attempt running ASR multiple times on failure

### PR DESCRIPTION
**Pull request by howiekaye**
Run ASR multiple times on failure.  The number of attempts is controlled by the ASRAttempts configuration setting in the com.google.corp.restor profile.

### CODE REVIEW

**Comment by erikng** [comment]
You should update the readme with the new key as well.

File: restord/ImageSessionServer.m [hunk 1]
```diff
@@ -260,39 +260,50 @@ - (NSString *)apfsOSDisk {
...
-  // Launch and wait for exit.
-  [self.asr launch];
-  [self.asr waitUntilExit];
-
-  // Clear readability handler or the file handle is never released.
-  outputFh.readabilityHandler = nil;
+  int retries = (int)[[[NSUserDefaults alloc] initWithSuiteName:kPreferenceDomain] integerForKey:kASRAttempts];
+  if (retries <= 0) retries = 1;
+  NSLog(@"Will attempt ASR up to %d times.", retries);
+  for(int i = 0; i < retries; i++) {
```
Comment (tburgin) [inline restord/ImageSessionServer.m]: nit: Missing space after for
nit: Pre-increment i 
for (int i = 0; i < retries; ++i) {

File: restord/ImageSessionServer.m [hunk 2]
```diff
@@ -260,39 +260,50 @@ - (NSString *)apfsOSDisk {
...
+      [self processOutput:(NSData *)availableData];
+    };
+
+    // Launch and wait for exit.
+    [self.asr launch];
+    [self.asr waitUntilExit];
+
+    // Clear readability handler or the file handle is never released.
+    outputFh.readabilityHandler = nil;
+    if (self.asr && self.asr.terminationStatus == 0) {
```
Comment (tburgin) [inline restord/ImageSessionServer.m]: nit: if (self.asr && self.asr.terminationStatus == 0) break;

File: restord/ImageSessionServer.m (hunk 1 above)
Comment (howiekaye) [reply restord/ImageSessionServer.m]: Done.  I don't think the compiler cares about pre-increment vs post.

File: restord/ImageSessionServer.m (hunk 2 above)
Comment (howiekaye) [reply restord/ImageSessionServer.m]: Done.

**Comment by howiekaye** [comment]
PTAL

File: restord/ImageSessionServer.m (hunk 1 above)
Comment (tburgin) [reply restord/ImageSessionServer.m]: I think compilers care very much about them. I think the point you are trying to make is that, in the end the compiler optimizes away the additional instructions from i++. If so, I agree but choose to be explicit.

**Review by tburgin — APPROVED** [review]

**Comment by howiekaye** [comment]
Yes, that is what I meant.  Please go ahead and merge.

**Pull request merged**